src/internal/m365/service/groups/export.go

```diff
@@ -17,20 +17,34 @@ import (
 	"github.com/alcionai/corso/src/pkg/fault"
 	"github.com/alcionai/corso/src/pkg/logger"
 	"github.com/alcionai/corso/src/pkg/path"
+	"github.com/alcionai/corso/src/pkg/services/m365/api"
+	"github.com/alcionai/corso/src/pkg/services/m365/api/graph"
 )
 
-var _ inject.ServiceHandler = &baseGroupsHandler{}
+var _ inject.ServiceHandler = &groupsHandler{}
 
 func NewGroupsHandler(
 	opts control.Options,
-) *baseGroupsHandler {
-	return &baseGroupsHandler{
-		opts:               opts,
-		backupDriveIDNames: idname.NewCache(nil),
-		backupSiteIDWebURL: idname.NewCache(nil),
+	apiClient api.Client,
+	resourceClient idname.GetResourceIDAndNamer,
```

Comment:
naming nit, I'd prefer Getter since this isn't a separate client in the way that api.Client is.  Makes it a little clearer what the expectations are for the complying struct.  IE: that it only needs to fulfill the methods; not that it maintains a connection to a resource.
```suggestion
	resourceGetter idname.GetResourceIDAndNamer,
```